chapter2_sites/chapter2_sites.md

```diff
@@ -0,0 +1,310 @@
+
+#### Run the Script in Debug
+Let’s increase the logging output in the example to ```DEBUG``` to understand what’s happening in more detail. See the Log Level section in Chapter 1 on logging configuration.
+
+```
+> 2021-09-16 14:29:19,213 connectionpool DEBUG _new_conn:   Starting new HTTPS connection (1): us-qa-api.sitelink.topcon.com:443
+````
+
+The script exchanges the OAuth credentials for a token using the ```oauth``` microservice.
+
+```
+> 2021-09-16 14:29:20,702 connectionpool DEBUG _make_request:   https://us-qa-api.sitelink.topcon.com:443 "POST /oauth/v1/token?grant_type=client_credentials&scope=9f2337a2400adf146ff7629c0a391079 HTTP/1.1" 200 372
+```
+The script prints a header line confirming the example running and its configuration.
+```
+> 2021-09-16 14:29:20,706 site_create INFO main:   Running site_create.py for server=https://us-qa-api.sitelink.topcon.com:443 dc=us owner=ce235e5e-6d87-4a84-80f2-0e56b137a132
+```
+The script posts a site creation frame to the ```siteowner``` microservice running in the QA environment. The owner ID is provided in the URL. The frame contains the following for the new site:
+-	A uuid string uniquely identifying the site. Note that this is not the same as the site identifier that is returned by ```siteowner``` when the site is created.
+-	A name to represent the site. Site names need not be unique.
+-	The data center to create the site in. Best practice is to select the data center closest to the site. Once the site is created, the data center cannot be changed. 
+-	An AsBuilt configuration determined by work type:
+o	Small: Suitable for sites that mostly involve paving and curbing with a typical machine operating speed of 4 – 16 km/h.
+o	Medium: Suitable for sites that involve bulk earth works, paving and compaction with a typical machine operating speed of 6 – 22 km/h. 
+o	Large: Suitable for piste management with a typical machine operating speed of 10 – 42 km/h.
+
+```
+> 2021-09-16 14:29:20,707 site_create INFO create_site:   post site creation to site owner https://us-qa-api.sitelink.topcon.com:443/siteowner/v1/owners/ce235e5e-6d87-4a84-80f2-0e56b137a132/create_site
+> 2021-09-16 14:29:20,708 site_create DEBUG create_site:   {
+    "site_uuid": "e3f82b0c-f8b3-41c0-9fe8-f654c4b4662a",
+    "name": "Example Site",
+    "dc": "us",
+    "region": "medium"
+}
+```
+
+The ```siteowner``` microservice responds with a ```201``` and the following frame:
+-	_id, rev, type are RDM concepts covered in Chapter 3.
+-	Identifier is the string that will be used to reference this site in all future calls.
+
+```
+> 2021-09-16 14:29:21,047 connectionpool DEBUG _make_request:   https://us-qa-api.sitelink.topcon.com:443 "POST /siteowner/v1/owners/ce235e5e-6d87-4a84-80f2-0e56b137a132/create_site 
```

Comment:
if it's possible to add colour code to these snippets it would look nice and make it more readable